Mathlib/CategoryTheory/ChosenFiniteProducts.lean

```diff
@@ -0,0 +1,126 @@
+/-
+Copyright (c) 2024 Adam Topaz All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+
+import Mathlib.CategoryTheory.Monoidal.OfChosenFiniteProducts.Symmetric
+import Mathlib.CategoryTheory.Limits.Constructions.FiniteProductsOfBinaryProducts
+
+/-!
+
+# Categories with chosen finite products
+
+We introduce a class, `ChosenFiniteProducts`, which bundles explicit choices
+for a terminal object and binary products in a category `C`.
+This is primarily useful for categories which have finite products with good
+definitional properties, such as the category of types.
```

Comment:
I care more about the category of schemes, or affine schemes, or affine schemes over Spec(R). If you are envisaging your work being helpful for these categories (which are much more interesting than the boring old category of types) then you could mention one or more of them here. If this is future work then it could be added as a TODO.

Reply:
Will do. Right now the category of types is the only example that's provided.